### PR TITLE
blender: fix blender wrapper mainProgram

### PR DESCRIPTION
**Pull request by lizelive**
## Description of changes
blender withPackages mainProgram is wrong.
could also be fixed by making wrapper not change the blender name.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mindavi — APPROVED** [review]

File: pkgs/applications/misc/blender/wrapper.nix [hunk 1]
```diff
@@ -8,7 +8,8 @@ stdenv.mkDerivation (finalAttrs: {
   pname = blender.pname + "-wrapped";
   src = blender;
 
-  inherit (blender) version meta;
+  inherit (blender) version;
+  meta = blender.meta // { mainProgram = pname; };
```
Comment (veprbl) [inline pkgs/applications/misc/blender/wrapper.nix:12]: I'm confused. How does this evaluate? There is no `pname` in scope, you'd needed to reference it as `finalAttrs.finalPackage.pname` or add `rec`/`with finalAttrs;` to make it work.

Comment (kirillrdy) [reply pkgs/applications/misc/blender/wrapper.nix:12]: it doesnt evaluate

```
blender.withPackages (p: [])
```

```

       at /home/kirillvr/.cache/nixpkgs-review/pr-268003/nixpkgs/pkgs/applications/misc/blender/wrapper.nix:12:42:

           11|   inherit (blender) version;
           12|   meta = blender.meta // { mainProgram = pname; };
             |                                          ^
           13|
```

**Comment by kirillrdy** [comment]
@lizelive when specifically mainProgram is not right ?

File: pkgs/applications/misc/blender/wrapper.nix (hunk 1 above)
Comment (lizelive) [reply pkgs/applications/misc/blender/wrapper.nix:12]: ah need to add rec

**Comment by kirillrdy** [comment]
@lizelive can you provide nix expression to reproduce the issue ?

**Comment by veprbl** [comment]
Please reopen once this is ready

**Pull request closed without merging**